src/Raven.Server/Documents/Handlers/BatchRequestParser.cs

```diff
@@ -418,17 +462,20 @@ private static unsafe CommandPropertyName GetPropertyType(JsonParserState state)
                     return CommandPropertyName.PatchIfMissing;
 
                 case 11:
-                    if (*(long*)state.StringBuffer != 7594869363257730379)
-                        return CommandPropertyName.NoSuchProperty;
-
-                    return CommandPropertyName.KeyPrefixed;
+                    if (*(long*)state.StringBuffer == 7594869363257730379)
```

Comment:
We are only comparing the first 8 bytes here, we need to compare the whole thing, just to be safe.